pulser-simulation/pulser_simulation/simulation.py

```diff
@@ -134,26 +130,10 @@ def __init__(
                     )
                 )
         _sampled_seq = replace(sampled_seq, samples_list=samples_list)
-        self._interaction = "XY" if _sampled_seq._in_xy else "ising"
         self._tot_duration = _sampled_seq.max_duration
         self.samples_obj = _sampled_seq.extend_duration(self._tot_duration + 1)
 
-        # Initializing qubit infos
-        self._qdict = self._register.qubits
-        self._size = len(self._qdict)
-        self._qid_index = {qid: i for i, qid in enumerate(self._qdict)}
-
-        # Type hints for attributes defined outside of __init__
-        self.basis_name: str
-        self._config: SimConfig
-        self.op_matrix: dict[str, qutip.Qobj]
-        self.basis: dict[str, qutip.Qobj]
-        self.dim: int
-        self._eval_times_array: np.ndarray
-        self._bad_atoms: dict[Union[str, int], bool] = {}
-        self._doppler_detune: dict[Union[str, int], float] = {}
-
-        # Initializing sampling and evalutaion times
+        # Testing sampling
         if not (0 < sampling_rate <= 1.0):
             raise ValueError(
                 "The sampling rate (`sampling_rate` = "
```

Comment:
I have hesitated in moving these tests to the init of Hamiltonian. What do you think of this ?